# Importamos librerias

In [44]:
import pandas as pd

Los csv del proyecto están guardados en la carpeta data/


In [ ]:
campaña = pd.read_csv('../data/campaña.csv')
clientes = pd.read_csv('../data/clientes.csv')
devoluciones = pd.read_csv('../data/devoluciones.csv')
lineas_pedido = pd.read_csv('../data/lineas_pedido.csv')
pedidos = pd.read_csv('../data/pedidos.csv')
productos = pd.read_csv('../data/productos.csv')
visitas_web = pd.read_csv('../data/visitas_web.csv')


# Exploración inicial de las base de datos

In [50]:
print('='*100)
print('Base de datos: Campaña')
print("="*100)
print(f'Shape : {campaña.shape}')   # (filas, columnas)
display(campaña.head())             # primeras 5 filas
print(campaña.dtypes)               # que tipo tiene cada columna
print(campaña.isnull().sum())       # Cuantos nulos hay por columnas

Base de datos: Campaña
Shape : (4, 8)


,campana_id,nombre,tipo,fecha_inicio,fecha_fin,canal,presupuesto,ventas_ano_anterior
0,1,Rebajas Verano 2024,rebajas,2024-07-01,2024-07-21,instagram,48000.0,512340.0
1,2,Rebajas Invierno 2024,rebajas,2024-01-07,2024-01-28,email,52000.0,498120.0
2,3,Black Friday 2023,black_friday,2023-11-24,2023-11-27,google,61.000 €,405780.0
3,4,Vuelta al Cole 2023,promo,2023-09-01,2023-09-15,instagram,22000.0,176900.0


campana_id               int64
nombre                     str
tipo                       str
fecha_inicio               str
fecha_fin                  str
canal                      str
presupuesto                str
ventas_ano_anterior    float64
dtype: object
campana_id             0
nombre                 0
tipo                   0
fecha_inicio           0
fecha_fin              0
canal                  0
presupuesto            0
ventas_ano_anterior    0
dtype: int64


## Tabla campaña
- Shape : (4, 8)
- campana_id : identificador único de la tabla camapañas. Es la clave: no se repite (salvo los duplicados de exportación que hay que quitar).
- fecha_inicio, fecha_fin: Fecha de inicio y fecha final de la campaña. Ojo: Formato de las columnas (str), hay que convertir a formato fecha. 
- presupuesto: Capital destinado para la campaña. Ojo: hay signos de moneda en ciertos registros, hay que convertir a número.

In [51]:
print('='*100)
print('Base de datos: Clientes')
print("="*100)
print(f'Shape : {clientes.shape}')
display(clientes.head())
print(clientes.dtypes)
print(clientes.isnull().sum())

Base de datos: Clientes
Shape : (5010, 5)


,cliente_id,fecha_registro,ciudad,genero,edad
0,2371,2019-10-18,bilbao,F,29
1,3537,2021-10-14,Bilbao,M,37
2,2088,2023-10-05,madrid,m,36
3,2769,2018-01-01,Lisboa,F,36
4,1567,2021-06-25,Madrid,M,42


cliente_id        int64
fecha_registro      str
ciudad              str
genero              str
edad              int64
dtype: object
cliente_id          0
fecha_registro    150
ciudad              0
genero              0
edad                0
dtype: int64


## Tabla clientes
- Shape : (5010, 5)
- cliente_id: identificar de cliente. Número entero. Es clave: cada cliente tiene id único. 
- fecha_registro:  Fecha de registro del cliente. Ojo: Formato de las columnas (str), hay que convertir a formato fecha. Existen 150 registros nulos. 
- ciudad: Ciudad donde se hizo el registro o lugar donde recide el cliente. Ojo: Texto y viene sucia: "bilbabo", "Bilbao". Hay que normalizar. 
- genero: género del cliente. Ojo: Existen variaciones. Estandarizar la columna para que no existan categorías adicionales a F y M.  

In [52]:
print('='*100)
print('Base de datos: devoluciones')
print("="*100)
print(f'Shape : {devoluciones.shape}')
display(devoluciones.head())
print(devoluciones.dtypes)
print(devoluciones.isnull().sum())

Base de datos: devoluciones
Shape : (5558, 5)


,devolucion_id,linea_id,fecha_devolucion,talla,motivo
0,702838,910363,15/07/2024,36,talla incorrecta
1,700079,900324,18/07/2024,XL,no me gusta
2,704821,917455,2024-07-16,m,talla incorrecta
3,701713,906266,17/07/2024,s,no me gusta
4,701621,905938,2024-07-05,36,talla incorrecta


devolucion_id       int64
linea_id            int64
fecha_devolucion      str
talla                 str
motivo                str
dtype: object
devolucion_id       0
linea_id            0
fecha_devolucion    0
talla               0
motivo              0
dtype: int64


## Tabla devoluciones
- Shape : (5558, 5)
- devolucion_id: identificador único de la devolución. 
- linea_id:  identificador único de la línea. Número entero. Es la clave: no se repite. 
- fecha_devolución : Fecha de la devolución. Ojo: Formato de las columnas (str), hay que convertir a formato fecha.
- talla: talla comprada. Texto, y viene sucia: "M", "m", "Talla M" son la misma. Hay que normalizar.
- motivo: motivo de la devolución. 


In [48]:
print('='*100)
print('Base de datos: lineas pedido')
print("="*100)
print(f'Shape : {lineas_pedido.shape}')
display(lineas_pedido.head())
print(lineas_pedido.dtypes)
print(lineas_pedido.isnull().sum())

Base de datos: lineas pedido
Shape : (20140, 7)


,linea_id,pedido_id,producto_id,talla,cantidad,precio_unitario,descuento_pct
0,900001,500001,1137,39,1,"63,83 €",0
1,900002,500002,1232,XL,1,"26,48 €",0.3
2,900003,500002,1028,L,2,"48,82 €",30%
3,900004,500003,1229,S,1,"38,72 €",0.0
4,900005,500003,1284,S,2,"20,70 €",0%


linea_id           int64
pedido_id          int64
producto_id        int64
talla                str
cantidad           int64
precio_unitario      str
descuento_pct        str
dtype: object
linea_id           0
pedido_id          0
producto_id        0
talla              0
cantidad           0
precio_unitario    0
descuento_pct      0
dtype: int64


## Tabla lineas pedido
- Shape : (20140, 7)
- linea_id :  identificador único de la línea
- pedido_id :  a qué pedido pertenece la línea. Número entero. Apunta a la tabla pedidos.
- producto_id : qué producto es. Número entero. Apunta a la tabla productos.
- talla: talla comprada. Texto, y viene sucia: "M", "m", "Talla M" son la misma. Hay que normalizar.
- precio_unitario:  precio de catálogo por unidad. Viene como texto con "€" y coma decimal ("29,99 €"): hay que convertir a número.
- descuento_pct :  descuento aplicado. Viene en tres formatos ("20%", 0.2, 20): hay que normalizar a fracción 0-1.
- Cantidad: Cantidad comprada. Ojo: 6 valores negativos. 

In [47]:
print('='*100)
print('Base de datos: pedidos')
print("="*100)
print(f'Shape : {pedidos.shape}')
display(pedidos.head())
print(pedidos.dtypes)
print(pedidos.isnull().sum())

Base de datos: pedidos
Shape : (8000, 7)


,pedido_id,cliente_id,fecha_pedido,campana_id,canal,ciudad,metodo_pago
0,500001,2290,24/06/2024,NaN,directo,Oporto,tarjeta
1,500002,4937,2024-07-02,1.0,instagram,lisboa,tarjeta
2,500003,544,2024-06-27,NaN,email,Oporto,bizum
3,500004,34,2024-07-12,1.0,email,Zaragoza,paypal
4,500005,4507,2024-07-05,1.0,instagram,Madrid,paypal


pedido_id         int64
cliente_id        int64
fecha_pedido        str
campana_id      float64
canal               str
ciudad              str
metodo_pago         str
dtype: object
pedido_id          0
cliente_id         0
fecha_pedido       0
campana_id      1179
canal              0
ciudad             0
metodo_pago        0
dtype: int64


## Tabla pedidos
- Shape : (8000, 7)
- campana_id : identificador único de la tabla camapañas. Ojo: Existen 1179 valores nulos identificados. 
- fecha_pedido: fecha en la que se hizo el pedido. Ojo: formato incorrecto, hay que convertir a formato fecha. 
- ciudad: Ciudad donde se hizo el registro o lugar donde recide el cliente. Ojo: Texto y viene sucia: "bilbabo", "Bilbao". Hay que normalizar. 

In [46]:
print('='*100)
print('Base de datos: productos')
print("="*100)
print(f'Shape : {productos.shape}')
display(productos.head())
print(productos.dtypes)
print(productos.isnull().sum())

Base de datos: productos
Shape : (400, 7)


,producto_id,nombre,categoria,temporada,coleccion,coste_unitario,precio_base
0,1001,Vestido Aura,Vestidos,Verano24,Cápsula Sol,12.47,32.57
1,1002,Vestido Iris,VESTIDOS,verano,Atelier,11.01,27.23
2,1003,Vestido Nácar,vestido,verano 2024,Mediterráneo,22.75,55.06
3,1004,Vestido Junio,vestido,Verano24,Básicos,21.83,45.55
4,1005,Vestido Bruma,vestido,verano,Denim,12.20,31.32


producto_id         int64
nombre                str
categoria             str
temporada             str
coleccion             str
coste_unitario    float64
precio_base       float64
dtype: object
producto_id        0
nombre             0
categoria          0
temporada          0
coleccion          0
coste_unitario    12
precio_base        0
dtype: int64


## Tabla productos
- Shape : (400, 7)
- categoria: Categeoría del producto. Ojo: Datos sucios ' Vestidos',     'VESTIDOS',      'vestido',     'vestidos', hay que normalizar. 
- temporada: Temporada en la que se vendio el producto. Ojo: Datos sucios 'Verano24', 'verano', 'verano 2024', 'SS24', 'Verano 2024', hay que normalizar. 
- coste_unitario: Costo unitario del producto. Ojo:12 valores nulos encontrados en la columna coste_unitario

In [45]:
print('='*100)
print('Base de datos: visitas web')
print("="*100)
print(f'Shape : {visitas_web.shape}')
display(visitas_web.head())
print(visitas_web.dtypes)
print(visitas_web.isnull().sum())

Base de datos: visitas web
Shape : (112, 3)


,fecha,canal,sesiones
0,2024-06-24,email,NaN
1,2024-06-24,instagram,1421.0
2,2024-06-24,google,1078.0
3,2024-06-24,directo,471.0
4,2024-06-25,email,307.0


fecha           str
canal           str
sesiones    float64
dtype: object
fecha       0
canal       0
sesiones    4
dtype: int64


## Tabla visitas_web
- Shape : (112, 3)
- fecha: fecha de visita al sitio web. Ojo: Formato incorrecto (str), convertir a formato fecha 
- canal: medio por donde nos visita el cliente. 
- sesiones: representa el número total de interacciones o visitas continuas que los usuarios realizan en tu sitio web dentro de un periodo determinado. Ojo: Existen 4 valores nulos.


# Análisis General

## Tabla campaña
- Shape : (4, 8)
- campana_id : identificador único de la tabla camapañas. Es la clave: no se repite (salvo los duplicados de exportación que hay que quitar).
- fecha_inicio, fecha_fin: Fecha de inicio y fecha final de la campaña. Ojo: Formato de las columnas (str), hay que convertir a formato fecha. 
- presupuesto: Capital destinado para la campaña. Ojo: hay signos de moneda en ciertos registros, hay que convertir a número.

## Tabla clientes
- Shape : (5010, 5)
- cliente_id: identificar de cliente. Número entero. Es clave: cada cliente tiene id único. 
- fecha_registro:  Fecha de registro del cliente. Ojo: Formato de las columnas (str), hay que convertir a formato fecha. Existen 150 registros nulos. 
- ciudad: Ciudad donde se hizo el registro o lugar donde recide el cliente. Ojo: Texto y viene sucia: "bilbabo", "Bilbao". Hay que normalizar. 
- genero: género del cliente. Ojo: Existen variaciones. Estandarizar la columna para que no existan categorías adicionales a F y M.  

## Tabla devoluciones
- Shape : (5558, 5)
- devolucion_id: identificador único de la devolución. 
- linea_id:  identificador único de la línea. Número entero. Es la clave: no se repite. 
- fecha_devolución : Fecha de la devolución. Ojo: Formato de las columnas (str), hay que convertir a formato fecha.
- talla: talla comprada. Texto, y viene sucia: "M", "m", "Talla M" son la misma. Hay que normalizar.
- motivo: motivo de la devolución. Ojo: Hay dos categorías que dicen los mismo "talla incorrecta", "Talla". 


## Tabla lineas pedido
- Shape : (20140, 7)
- linea_id :  identificador único de la línea
- pedido_id :  a qué pedido pertenece la línea. Número entero. Apunta a la tabla pedidos.
- producto_id : qué producto es. Número entero. Apunta a la tabla productos.
- talla: talla comprada. Texto, y viene sucia: "M", "m", "Talla M" son la misma. Hay que normalizar.
- precio_unitario:  precio de catálogo por unidad. Viene como texto con "€" y coma decimal ("29,99 €"): hay que convertir a número.
- descuento_pct :  descuento aplicado. Viene en tres formatos ("20%", 0.2, 20): hay que normalizar a fracción 0-1.

## Tabla pedidos
- Shape : (8000, 7)
- campana_id : identificador único de la tabla camapañas. Ojo: Existen 1179 valores nulos identificados. 
- fecha_pedido: fecha en la que se hizo el pedido. Ojo: formato incorrecto, hay que convertir a formato fecha. 
- ciudad: Ciudad donde se hizo el registro o lugar donde recide el cliente. Ojo: Texto y viene sucia: "bilbabo", "Bilbao". Hay que normalizar. 

## Tabla productos
- Shape : (400, 7)
- categoria: Categeoría del producto. Ojo: Datos sucios ' Vestidos',     'VESTIDOS',      'vestido',     'vestidos', hay que normalizar. 
- temporada: Temporada en la que se vendio el producto. Ojo: Datos sucios 'Verano24', 'verano', 'verano 2024', 'SS24', 'Verano 2024', hay que normalizar. 
- coste_unitario: Costo unitario del producto. Ojo:12 valores nulos encontrados en la columna coste_unitario

## Tabla visitas_web
- Shape : (112, 3)
- fecha: fecha de visita al sitio web. Ojo: Formato incorrecto (str), convertir a formato fecha 
- canal: medio por donde nos visita el cliente. 
- sesiones: representa el número total de interacciones o visitas continuas que los usuarios realizan en tu sitio web dentro de un periodo determinado. Ojo: Existen 4 valores nulos.
